lib/rpmsg/rpmsg_virtio.c

```diff
@@ -288,10 +288,16 @@ static int _rpmsg_virtio_get_buffer_size(struct rpmsg_virtio_device *rvdev)
 
 static void rpmsg_virtio_hold_rx_buffer(struct rpmsg_device *rdev, void *rxbuf)
 {
+	struct rpmsg_virtio_device *rvdev;
 	struct rpmsg_hdr *rp_hdr;
 
-	(void)rdev;
+	rvdev = metal_container_of(rdev, struct rpmsg_virtio_device, rdev);
 
+	if ((rxbuf < ((struct metal_io_region *)rvdev->rvq->shm_io)->virt + sizeof(struct rpmsg_hdr)) ||
+	    (rxbuf >= (((struct metal_io_region *)rvdev->rvq->shm_io)->virt +
+	    ((struct metal_io_region *)rvdev->rvq->shm_io)->size)))
+		return;
+		
```

Comment:
The metal_io_region structure should be manipulated in libmetal.
please create metal_io_block_check API in libmetal instead